lib/amazonka/src/Amazonka/Auth/ConfigFile.hs

```diff
@@ -162,7 +162,7 @@ mergeConfigs creds confs =
     (HashMap.fromList <$> stripProfiles confs)
   where
     stripProfiles :: HashMap Text v -> HashMap Text v
-    stripProfiles = HashMap.mapKeys $ Text.unwords . stripProfile . Text.words
+    stripProfiles = HashMap.fromList . map (\(k, v) -> (Text.unwords $ stripProfile $ Text.words k, v)) . HashMap.toList
```

Comment:
do we have common keys after appying the stripProfile?

Reply:
```
    stripProfile = \case
      [w] -> [w]
      ("profile" : ws) -> ws
      ws -> ws
```

we might have as per stripProfile function def